gui/widgets/editable.cpp

```diff
@@ -63,20 +63,32 @@ void EditableWidget::reflowLayout() {
 }
 
 void EditableWidget::setEditString(const String &str) {
-	// TODO: We probably should filter the input string here,
-	// e.g. using tryInsertChar.
-	_editString = str;
+	_editString = "";
+
+	// Copy string char by char, ensuring each char is valid
+	uint len = 0;
+	for (uint i = 0; i < str.size() && len < _maxLength; i++)
+		if (isCharValid((byte)str[i])) {
+			_editString += str[i];
+			len++;
+		}
+
 	_caretPos = 0;
 }
 
 bool EditableWidget::tryInsertChar(byte c, int pos) {
-	if ((c >= 32 && c <= 127) || c >= 160) {
+	if (_maxLength == 0 || _editString.size() < _maxLength) {
 		_editString.insertChar(c, pos);
 		return true;
 	}
+
 	return false;
 }
 
+bool EditableWidget::isCharValid(byte c) {
```

Comment:
This is probably a good candiate for a method to be made constant because it should not modify internal state of the widget.